crypto/rand/rand_lib.c

```diff
@@ -228,11 +228,9 @@ static uint64_t get_timer_bits(void)
     }
 #else
 
-# if defined(_POSIX_C_SOURCE) \
+# if (defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 199309L) \
      && defined(_POSIX_TIMERS) \
-     && _POSIX_C_SOURCE >= 199309L \
-     && (!defined(__GLIBC__) \
-         || (defined(__GLIBC_PREREQ) && __GLIBC_PREREQ(2, 17)))
+     && (!defined(__GLIBC__) || __GLIBC_PREREQ(2, 17))
```

Comment:
I'd say it deserves a note *why* is it used.

On side note one can also wonder if one should learn to occasionally say no. I mean when something is that intricate, maybe it's appropriate to ask what is the gain really, and if it worth the trouble...

Reply:
I for one don't see why this whole business of `__GLIBC_PREREQ` came up, and would much prefer just removing it.

Reply:
Older versions of glibc had the clock_gettime functionality in a separate library (librt), newer versions have it in glibc.  We don't want to add library dependencies before the next release because they're considered an API change but we'd like the functionality.  Hence the version checking